src/brsTypes/components/RoSGNode.ts

```diff
@@ -265,6 +277,60 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         return BrsInvalid.Instance;
     }
 
+    private removeChildByReference(child: BrsType): boolean {
+        if (child instanceof RoSGNode) {
+            let spliceIndex = this.children.indexOf(child);
+            if (spliceIndex >= 0) {
+                child.removeParent();
+                this.children.splice(spliceIndex, 1);
+            }
+            return true;
+        }
+        return false;
+    }
+
+    private appendChildToParent(child: BrsType): boolean {
+        if (child instanceof RoSGNode) {
+            if (this.children.includes(child)) {
+                return true;
+            }
+            this.children.push(child);
+            child.setParent(this);
+            return true;
+        }
+        return false;
+    }
+
+    private replaceChildAtIndex(newchild: BrsType, index: Int32): boolean {
+        let childrenSize = this.children.length;
+        let indexValue = index.getValue();
+        if (newchild instanceof RoSGNode && indexValue < childrenSize) {
+            this.removeChildByReference(newchild);
+            if (indexValue >= 0) {
+                if (indexValue < this.children.length) {
```

Comment:
The method replaces a child at the index given with a new child. If the new child already exists, it removes the child and then does the replacing process since you can't have two of the same child. I can explain more in person if you want

Reply:
thanks for explaining offline, do you mind adding some comments to the `this.removeChildByReference(newchild);` line?

Reply:
Yep, can do